web/src/containers/ExecutiveSummary.js

```diff
@@ -16,106 +16,103 @@ import {
   selectBudgetGrandTotal
 } from '../reducers/budget.selectors';
 
-class ExecutiveSummary extends PureComponent {
-  jump = to => () => {
-    const { jumpTo: action } = this.props;
-    action(to);
-  };
-
-  render() {
-    const { data, total, years } = this.props;
-    return (
-      <Waypoint id="executive-summary-overview">
-        <Section isNumbered id="executive-summary" resource="executiveSummary">
-          <Waypoint id="executive-summary-summary" />
-          <Subsection
-            id="executive-summary-summary"
-            resource="executiveSummary.summary"
-          >
-            {data.map((activity, i) => (
-              <Review
-                key={activity.key}
-                heading={
-                  <Fragment>
-                    Activity {i + 1}:{' '}
-                    <a
-                      href={`#activity-${activity.key}`}
-                      onClick={this.jump(activity.key)}
-                    >
-                      {activity.name || t('activities.noNameYet')}
-                    </a>
-                  </Fragment>
-                }
-                headingLevel={4}
-                editHref={`#activity-${activity.key}`}
-                onEditClick={this.jump(activity.key)}
-                className={i === data.length - 1 ? 'ds-u-border-bottom--0' : ''}
-              >
-                {activity.summary && <p>{activity.summary}</p>}
+import ActivityDialog from './activity/EntryDetailsDialog.js';
 
-                <ul className="ds-c-list--bare">
-                  <li>
-                    <strong>Date:</strong> {activity.dateRange}
-                  </li>
-                  <li>
-                    <strong>Total cost of activity:</strong>{' '}
-                    <Dollars long>{activity.combined}</Dollars>
-                  </li>
-                  <li>
-                    <strong>Medicaid share:</strong>{' '}
-                    <Dollars long>{activity.medicaid}</Dollars> (
-                    <Dollars long>{activity.federal}</Dollars> Federal share)
-                  </li>
-                </ul>
-              </Review>
-            ))}
-            <hr className="ds-u-border--dark ds-u-margin--0" />
+const ExecutiveSummary = props => {
+  const [showModal, setShowModal] = useState(false);
+  const [activityIndexForModal, setActivityIndexForModal] = useState(-1);
+  const { data, total, years } = props;
+  const openModal = index => {
+    setActivityIndexForModal(index);
+    setShowModal(true);
+  };
```

Comment:
@mgwalker I could use some debugging help, if you have a chance. For some reason, even though this fires on every `Edit` click, it only opens a modal the first time you click edit. Once you've opened one activity modal, you can't open any more. There aren't any console errors that I can see, and it seems like it's firing this but not showing the`ActivityDialog` component on line 69. I'm pretty baffled, and all I can think is that maybe setting the state like this doesn't work like I think it should?

Reply:
`useState` is for component-level state, not application-level state, so the `ExecutiveSummary` and `EntryDetailsDialog` components each have their own `showModal` state. When you click "edit activity", both of them are true. When you close the modal, the `EntryDetailsDialog.showModal` becomes false, so it gets hidden. The problem is that there's no way to set it true again.

I think there either needs to be application-level state *or* the `EntryDetailsDialog` needs something like an `onClose()` prop that you can subscribe to, and set `ExecutiveSummary.showModal` to false when it's called. Then you could do away with the `showModal` state in `EntryDetailsDialog` altogether.

Reply:
All the facepalms. Of course, that makes sense!

> the EntryDetailsDialog needs something like an onClose() prop that you can subscribe to
This seems like a better way to handle visual changes than storing that information in app-level state, but I'm not sure how to go about that. Are we doing it anywhere in the current codebase that I could look at?